template-manager-core/src/main/java/com/sequenceiq/cloudbreak/template/views/provider/RdsViewProvider.java

```diff
@@ -0,0 +1,141 @@
+package com.sequenceiq.cloudbreak.template.views.provider;
+
+import java.util.Locale;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nonnull;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.DatabaseVendor;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.database.base.DatabaseType;
+import com.sequenceiq.cloudbreak.domain.RDSConfig;
+import com.sequenceiq.cloudbreak.domain.RdsSslMode;
+import com.sequenceiq.cloudbreak.domain.view.RdsConfigWithoutCluster;
+import com.sequenceiq.cloudbreak.template.views.RdsView;
+import com.sequenceiq.cloudbreak.template.views.dialect.DefaultRdsViewDialect;
+import com.sequenceiq.cloudbreak.template.views.dialect.OracleRdsViewDialect;
+import com.sequenceiq.cloudbreak.template.views.dialect.RdsViewDialect;
+import com.sequenceiq.cloudbreak.template.views.dialect.ServiceIdOracleRdsViewDialect;
+
+@Component
+public class RdsViewProvider {
+
+    public static final String WITHOUT_JDBC_PREFIX_REGEX = "^(.*?):(\\d*)[:/]?(\\w+)?";
+
+    private static final Pattern WITHOUT_JDBC_PREFIX_PATTERN = Pattern.compile(WITHOUT_JDBC_PREFIX_REGEX);
+
+    private static final String SSL_OPTIONS_WITHOUT_CERTIFICATE_FILE_PATH = "sslmode=verify-full&sslrootcert=";
+
+    private static final int HOST_GROUP_INDEX = 1;
+
+    private static final int PORT_GROUP_INDEX = 2;
+
+    private static final int DATABASE_GROUP_INDEX = 3;
+
+    public RdsView getRdsView(RdsConfigWithoutCluster rdsConfig) {
+        return getRdsView(rdsConfig, "");
+    }
+
+    public RdsView getRdsView(RdsConfigWithoutCluster rdsConfigWithoutCluster, String sslCertificateFilePath) {
+        RDSConfig rdsConfig = new RDSConfig();
+        rdsConfig.setArchived(rdsConfigWithoutCluster.isArchived());
+        rdsConfig.setConnectionDriver(rdsConfigWithoutCluster.getConnectionDriver());
+        rdsConfig.setConnectionURL(rdsConfigWithoutCluster.getConnectionURL());
+        rdsConfig.setConnectionPassword(rdsConfigWithoutCluster.getConnectionPassword());
+        rdsConfig.setConnectionUserName(rdsConfigWithoutCluster.getConnectionUserName());
+        rdsConfig.setConnectorJarUrl(rdsConfigWithoutCluster.getConnectorJarUrl());
+        rdsConfig.setCreationDate(rdsConfigWithoutCluster.getCreationDate());
+        rdsConfig.setDatabaseEngine(rdsConfigWithoutCluster.getDatabaseEngine());
+        rdsConfig.setDeletionTimestamp(rdsConfigWithoutCluster.getDeletionTimestamp());
+        rdsConfig.setDescription(rdsConfigWithoutCluster.getDescription());
+        rdsConfig.setId(rdsConfigWithoutCluster.getId());
+        rdsConfig.setName(rdsConfigWithoutCluster.getName());
+        rdsConfig.setSslMode(rdsConfigWithoutCluster.getSslMode());
+        rdsConfig.setType(rdsConfigWithoutCluster.getType());
```

Comment:
I know this was introduced earlier, but perhaps it would be nicer to extract the logic into a factory method `static RDSConfig RDSConfig.fromRdsConfigWithoutCluster(RdsConfigWithoutCluster rdsConfigWithoutCluster)` (or perhaps as `RDSConfig RdsConfigWithoutCluster.toRDSConfig()`).